src/fsharp/FSharp.Core/array.fs

```diff
@@ -478,6 +478,17 @@ namespace Microsoft.FSharp.Collections
         [<CompiledName("Where")>]
         let where f (array: _[]) = filter f array
 
+        [<CompiledName("Except")>]
+        let except (array1:'T[]) (array2:'T[]) =
+            checkNonNull "array1" array1
+            checkNonNull "array2" array2
+
+            if array1.Length = 0 then
```

Comment:
I would agree it is surprising, I had originally implemented it the way you assumed it was implemented. It was only when performance testing and the F# implementation was so much slower than the LINQ that I looked at the LINQ implementation more closely and noticed it.